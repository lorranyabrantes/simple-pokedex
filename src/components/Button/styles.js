import { StyleSheet } from "react-native";
import { COLORS, SPACING, BORDER_RADIUS } from "../../services/styles";

const styles = StyleSheet.create({
  default: {
    backgroundColor: COLORS.YELLOW,
    height: 35,
    alignItems: "center",
    justifyContent: "center",
    borderRadius: BORDER_RADIUS.medium,
    paddingHorizontal: SPACING.medium,
  },
  disabled: {
    backgroundColor: COLORS.GRAY,
  },
});

export default styles;
