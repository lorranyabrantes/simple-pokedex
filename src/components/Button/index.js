import React from "react";
import { TouchableOpacity } from "react-native";

import Text from "../Text";

import styles from "./styles";

const Button = ({ text, disabled, loading, onPress }) => {
  return (
    <TouchableOpacity
      testID="button"
      style={[styles.default, disabled || loading ? styles.disabled : {}]}
      onPress={onPress}
      disabled={disabled || loading}
    >
      {text ? <Text>{text}</Text> : null}
    </TouchableOpacity>
  );
};

export default Button;
