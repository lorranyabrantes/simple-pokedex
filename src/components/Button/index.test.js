import React from "react";
import { render, fireEvent } from "@testing-library/react-native";

import Button from "./";

const props = {
  text: "Hello",
  disabled: false,
  loading: false,
  onPress: jest.fn(),
};

describe("Button", () => {
  test("should render correctly", async () => {
    const component = render(<Button {...props} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  test("expected the function to be called", async () => {
    const { getByTestId } = render(<Button {...props} />);

    fireEvent.press(getByTestId("button"));

    expect(props.onPress).toHaveBeenCalled();
  });
});
