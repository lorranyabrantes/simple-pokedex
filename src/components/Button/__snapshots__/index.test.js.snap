// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FECA1B",
      "borderRadius": 5,
      "height": 35,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 16,
    }
  }
  testID="button"
>
  <Text
    style={
      Array [
        Object {
          "color": "#333333",
          "fontFamily": "Roboto-Regular",
          "fontSize": 14,
        },
        undefined,
      ]
    }
  >
    Hello
  </Text>
</View>
`;
